File: producer/src/main/java/pl/polsl/hdised/producer/measurement/MeasurementService.java
package pl.polsl.hdised.producer.measurement;

import org.springframework.stereotype.Service;

@Service
public class MeasurementService {

    private MeasurementCreator measurementCreator;

    public MeasurementService(MeasurementCreator measurementCreator) {
        this.measurementCreator = measurementCreator;
    }

    public void startProducing(String topic) {
        this.measurementCreator.setProduce(true);
        Thread thread = new Thread(() -> {
            try {
                this.measurementCreator.StartProducingMeasurementsToDatabase(topic);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        });
        thread.start();
    }

    public void stopProducing() {
        this.measurementCreator.setProduce(false);
    }

}

File: golang-measurement-generator/measurementService.go
package main

type MeasurementService struct {
	MeasurementCreator *MeasurementCreator
}

func NewMeasurementService(creator *MeasurementCreator) *MeasurementService {
	return &MeasurementService{
		MeasurementCreator: creator,
	}
}

func (ms *MeasurementService) StartProducing(topic string) {
	ms.MeasurementCreator.SetProduce(true)
	go func() {
		ms.MeasurementCreator.StartProducingMeasurementsToDatabase(topic)
	}()
}

func (ms *MeasurementService) StopProducing() {
	ms.MeasurementCreator.SetProduce(false)
}
